obspy/signal/calibration.py

```diff
@@ -138,19 +138,25 @@ def rel_calib_stack(st1, st2, calib_file, window_len, overlap_frac=0.5,
     return freq, amp, phase
 
 
-def _calc_resp(calfile, nfft, sampfreq):
+def _calc_resp(calobj, nfft, sampfreq):
     """
     Calculate transfer function of known system.
 
-    :type calfile: str
-    :param calfile: file containing poles, zeros and scale factor for known
-        system
+    :type calobj: str or dict
+    :param calobj: dictionary or file name (gse2) containing poles, zeros and
+    sensitivity for known system
     :returns: complex transfer function, array of frequencies
     """
+    # read paz file if a file is specified
+    if isinstance(calobj, str):
```

Comment:
I would actually to the opposite and first check if its a dictionary and otherwise call the `read_paz()` function. Maybe that function will learn a few tricks in the future and then this will just work.